tensorflow/compiler/xla/service/gpu/launch_dimensions.cc

```diff
@@ -91,9 +91,13 @@ LaunchDimensions CalculateLaunchDimensions(const Shape& shape,
 
   int64 block_count = CeilOfRatio(num_elements, threads_per_block);
   if (few_waves) {
-    threads_per_block = std::min(threads_per_block, int64{128});
-    block_count = gpu_device_info.core_count *
-                  (gpu_device_info.threads_per_core_limit / threads_per_block);
+    int64 new_block_count =
+        gpu_device_info.core_count *
+        (gpu_device_info.threads_per_core_limit / threads_per_block);
```

Comment:
Should this line be instead `(gpu_device_info.threads_per_core_limit / 128)`? Can you add a comment regarding why or why not?

Reply:
You are right, there is something not right here.
Now I compute both new value first, then if needed, I update both of them at the same time.
This allow to keep the right dependency between the 2 variables.

I didn't add a comment as I'm not sure what to write. If you have an idea, tell me.

Reply:
Without few_waves, `block_count * threads_per_block` roughly equals to `num_elements`, which makes perfect sense - each thread handles one element. If `threads_per_block == new_threads_per_block`, `new_block_count * new_threads_per_block` is roughly `gpu_device_info.core_count * gpu_device_info.threads_per_core_limit`, which also makes sense, as it says "We want to cap the number of blocks to number of cores, and we want as many threads as possible under this limitation". When `threads_per_block != new_threads_per_block`, `new_block_count * new_threads_per_block` isn't obvious to me what it means.

Reply:
I do not understand your question:
"When threads_per_block != new_threads_per_block, new_block_count * new_threads_per_block isn't obvious to me what it means."

The code must eigheter change threads_per_block and block_count at the same time or none of them.

I amended the history to change the variable name (changed new_ to capped_) and added comments.

When we try to cap the number of blocks, we do not want to increase it. This is why we have the second if. This can happen for small number of elements.

Reply:
To me it makes more sense to define `capped_block_count` as:
`gpu_device_info.core_count * (gpu_device_info.threads_per_core_limit / capped_threads_per_block)`. Notice the change from `threads_per_block` to `capped_threads_per_block`. This way, the total number of threads is always maximized (given `number of blocks == gpu_device_info.core_count`).

Reply:
Good catch. This is what I intended to have. I amended the commit to divide by capped_threads_per_block and updated one test to match that.